### PR TITLE
SimpleOrbit updates

### PR DESCRIPTION
**Pull request by icweaver**
This PR updates the tests for `SimpleOrbit`, adds support for negative times, and re-defines when `z` is positive/negative to be in line with the new criteria introduced in ad996365887676ced6a90eb82a7bab1e72d623ea:
https://github.com/JuliaAstro/Transits.jl/blob/eaa263bd64de46c33d8e108909413aeee2818356/src/Transits.jl#L82-L91

### CODE REVIEW

**Comment by icweaver** [comment]
I'm currently setting `z` to 1 if the planet is on the half of its orbit nearest to us and 0 otherwise, but from a code re-use standpoint it would be nice if we could maintain the same condition that is only 1 while in transit. I will take a look at this next

**Comment by icweaver** [comment]
Huh, just tried it out in `exoplanet`, and apparently the clipping is not just on our end:
download

I was probably just getting too caught-up in trying to make everything look like our previous transit curves. Since `exoplanet` has a [test](https://github.com/exoplanet-dev/exoplanet/blob/9c84387cccf75c17d521e924aa5bde0cc29b543d/tests/orbits/simple_test.py#L53) comparing their simple orbit to a Keplerian one, I think we should be ok. I'll switch back to what we had before with the working version of `SimpleOrbit` we have now, and add in that test as well

File: src/orbits/simple.jl [hunk 1]
```diff
@@ -43,26 +45,47 @@ function relative_position(orbit::SimpleOrbit, t)
...
+
 function flip(orbit::SimpleOrbit, ror)
     t0 = orbit.t0 + orbit.half_period
     b = orbit.b / ror
     speed = orbit.speed / ror
     ref_time = orbit.t0
-    return SimpleOrbit(orbit.period, t0, b, orbit.duration, speed, orbit.half_period, ref_time)
+    return SimpleOrbit(
+        orbit.period, t0, b, orbit.duration, speed, orbit.half_period, ref_time
+    )
```
Comment (mileslucas) [inline src/orbits/simple.jl:67]:
```suggestion
    return SimpleOrbit(orbit.period, t0, b, orbit.duration, speed, orbit.half_period, ref_time)
```
no need to change the formatting here

File: src/orbits/simple.jl [hunk 2]
```diff
@@ -43,26 +45,47 @@ function relative_position(orbit::SimpleOrbit, t)
...
 
 function Base.show(io::IO, orbit::SimpleOrbit)
     T = orbit.duration
     P = orbit.period
     b = orbit.b
     t0 = orbit.t0
-    print(io, "SimpleOrbit(P=", P, ", T=", T, ", t0=", t0, ", b=", b, ")")
+    print(io,
+        "SimpleOrbit(P=$P, T=$T, t0=$t0, b=$b)"
+    )
```
Comment (mileslucas) [inline src/orbits/simple.jl:77]:
```suggestion
    print(io, "SimpleOrbit(P=$P, T=$T, t0=$t0, b=$b)")
```

File: src/orbits/simple.jl [hunk 3]
```diff
@@ -19,21 +19,23 @@ Circular orbit parameterized by the basic observables of a transiting system.
...
     half_period = 0.5 * period
     duration > half_period && error("duration cannot be longer than half the period")
-    speed = 2 * sqrt(1 - b^2) / duration
+    speed = 2.0 * sqrt(1.0 - b^2.0) / duration
     ref_time =  t0 - half_period
     # normalize time types
-    period, t0, duration, half_period, ref_time = promote(period, t0, duration, half_period, ref_time)
+    period, t0, duration, half_period, ref_time = promote(
+    period, t0, duration, half_period, ref_time
+    )
```
Comment (mileslucas) [inline src/orbits/simple.jl:30]:
```suggestion
	period, t0, duration, half_period, ref_time = promote(period, t0, duration, half_period, ref_time)
```

File: src/orbits/simple.jl [hunk 4]
```diff
@@ -19,21 +19,23 @@ Circular orbit parameterized by the basic observables of a transiting system.
...
     SimpleOrbit(period, t0, b, duration, speed, half_period, ref_time)
 end
 
 
 period(orbit::SimpleOrbit) = orbit.period
 duration(orbit::SimpleOrbit) = orbit.duration
 
-relative_time(orbit::SimpleOrbit, t) = (t - orbit.ref_time) % period(orbit) - orbit.half_period
+relative_time(orbit::SimpleOrbit, t) = mod(t - orbit.ref_time, orbit.period) - orbit.half_period
```
Comment (mileslucas) [inline src/orbits/simple.jl:38]: good catch!

**Review by mileslucas — CHANGES_REQUESTED** [review]
I'm a bit confused- besides the `mod` vs `%`/`rem` change, everything is all good, right?

The reason why you're seeing the cutoff is because that's at exactly 1/4 the period (right?). Exoplanet basically says "you can't have a duration > period/2, but we don't stop you from trying" and I think keeping consistency with them there is good enough. I only wrote the `SimpleOrbit` because I wanted to get the limb darkening curves started and didn't want to make Keplerian orbits yet :)

File: src/orbits/simple.jl [hunk 5]
```diff
@@ -19,21 +19,23 @@ Circular orbit parameterized by the basic observables of a transiting system.
 * `t0` - The midpoint time of the reference transit, similar units as `period`
 * `b` - The impact parameter of the orbit, unitless
 """
-function SimpleOrbit(;period, duration, t0=zero(period), b=0)
+function SimpleOrbit(;period, duration, t0=0.0, b=0.0)
     half_period = 0.5 * period
     duration > half_period && error("duration cannot be longer than half the period")
-    speed = 2 * sqrt(1 - b^2) / duration
+    speed = 2.0 * sqrt(1.0 - b^2.0) / duration
```
[the remark this answers is not in the document]
Comment (icweaver) [reply src/orbits/simple.jl:25]: I always wondered about that. Thanks!

**Comment by icweaver** [comment]
> I'm a bit confused- besides the `mod` vs `%`/`rem` change, everything is all good, right?
> 
> The reason why you're seeing the cutoff is because that's at exactly 1/4 the period (right?). Exoplanet basically says "you can't have a duration > period/2, but we don't stop you from trying" and I think keeping consistency with them there is good enough. I only wrote the `SimpleOrbit` because I wanted to get the limb darkening curves started and didn't want to make Keplerian orbits yet :)

Yea, I think that is reasonable! Definitely interested to compare to our `KeplerianOrbit` curves =]

**Review by icweaver — COMMENTED** [review]
It looks like our coverage dropped just a smidge. Working on that now

**Comment by icweaver** [comment]
Ok, things are looking green 👍🏾

**Review by mileslucas — APPROVED** [review]
Thanks @icweaver

**Pull request merged**